### ISSUE TITLE
Re-work basic metric scraping stream

### ISSUE THREAD

**Issue by chrisjl154** [Sicarius154/Watcher#2]
The basic HTTP processing stream implemented should receive a set of query targets, re-wire it such that the query targets passed are used in the stream.

We should also re-work the code so it is now Prometheus-specific, due to a change in how Watcher will work in future (we no longer need it to be generic, we can code specifically against Prometheus's HTTP API and our code should reflect that)

In short, the task list to complete is:

- [x] Create a case class to represent queries to use
- [x] Make use of a Seq[] of the aforementioned case classes
- [x] Re-wire the stream such that it now takes the connection details for Prometheus
- [x] Implement the functionality needed to serialise to/from JSON
- [x] Create a basic function using HTTP4S that takes a query string and returns the result
